crates/sui-framework/sources/display.move

```diff
@@ -140,61 +107,68 @@ module sui::display {
             i = i + 1;
         };
 
-        share(display)
+        display
+    }
+
+    // === Entry functions: Create ===
+
+    /// Create a new empty Display<T> object and keep it.
+    entry public fun create_and_keep<T: key>(pub: &Publisher, ctx: &mut TxContext) {
+        transfer(new<T>(pub, ctx), sender(ctx))
     }
 
     /// Manually bump the version and emit an event with the updated version's contents.
     entry public fun update_version<T: key>(
-        pub: &Publisher, d: &mut Display<T>
+        display: &mut Display<T>
     ) {
-        assert!(is_package<T>(pub), ENotOwner);
-        d.version = d.version + 1;
+        display.version = display.version + 1;
         event::emit(VersionUpdated<T> {
-            version: d.version,
-            fields: *&d.fields,
-            id: object::uid_to_inner(&d.id),
+            version: display.version,
+            fields: *&display.fields,
+            id: object::uid_to_inner(&display.id),
         })
     }
 
     // === Entry functions: Add/Modify fields ===
 
     /// Sets a custom `name` field with the `value`.
-    entry public fun add<T: key>(pub: &Publisher, d: &mut Display<T>, name: String, value: String) {
-        assert!(is_package<T>(pub), ENotOwner);
-        add_internal(d, name, value)
+    entry public fun add<T: key>(self: &mut Display<T>, name: String, value: String) {
+        add_internal(self, name, value)
     }
 
     /// Sets multiple `fields` with `values`.
     entry public fun add_multiple<T: key>(
-        pub: &Publisher, d: &mut Display<T>, fields: vector<String>, values: vector<String>
+        self: &mut Display<T>, fields: vector<String>, values: vector<String>
     ) {
         let len = vector::length(&fields);
-        assert!(is_package<T>(pub), ENotOwner);
         assert!(len == vector::length(&values), EVecLengthMismatch);
 
         let i = 0;
         while (i < len) {
-            add_internal(d, *vector::borrow(&fields, i), *vector::borrow(&values, i));
+            add_internal(self, *vector::borrow(&fields, i), *vector::borrow(&values, i));
             i = i + 1;
         };
     }
 
     /// Change the value of the field.
     /// TODO (long run): version changes;
-    entry public fun edit<T: key>(pub: &Publisher, d: &mut Display<T>, name: String, value: String) {
-        assert!(is_package<T>(pub), ENotOwner);
-        let (_k, _v) = vec_map::remove(&mut d.fields, &name);
-        add_internal(d, name, value)
+    entry public fun edit<T: key>(self: &mut Display<T>, name: String, value: String) {
+        let (_k, _v) = vec_map::remove(&mut self.fields, &name);
```

Comment:
```suggestion
        let  _ = vec_map::remove(&mut self.fields, &name);
```

Reply:
Turns out we can't use a single type when a tuple is returned. Compiler fails with an error. Reverting your suggestion.

Reply:
```
  error[E04005]: expected a single type
      ┌─ .../sui/crates/sui-framework/sources/display.move:156:14
      │
  156 │         let  _ = vec_map::remove(&mut self.fields, &name);
      │              ^ Invalid type for local
      │
      ┌─ .../sui/crates/sui-framework/sources/vec_map.move:52:69
      │
   52 │     public fun remove<K: copy, V>(self: &mut VecMap<K,V>, key: &K): (K, V) {
      │                                                                     ------ Expected a single type, but found expression list type: '((std=0x1)::string::String, (std=0x1)::string::String)'
```